@here/olp-sdk-dataservice-read/lib/cache/MetadataCacheRepository.ts

```diff
@@ -52,7 +52,8 @@ export class MetadataCacheRepository {
                     hrn,
                     layerId,
                     version: rq.getVersion(),
-                    partitionId: metadata.partition
+                    partitionId: metadata.partition,
+                    additionalFields: rq.getAdditionalFields()
```

Comment:
I don't think this is a good idea. Instead, i would consider updating the entry in the LRU. I.e. find the entry in the map and in case user specified any additional fields check if the requested fields are available in the entry, if not request new and update the metadata in the cache under the same key.

Reply:
updated, thanks.